0001-typedb/02-dev/02-schema.md

```diff
@@ -0,0 +1,948 @@
+
+access sub relation, 
+relates accessed-object, 
+relates valid-action, 
+plays change-request:requested-change;
+
+change-request sub relation, 
+relates requesting-subject, 
+relates requested-subject, 
+relates requested-change;
+```
+
+The example above defines one attribute type and five relation types:
+
+* `ownership` — subtypes the base `relation` type, is abstract, and relates `owned` and `owner` roles.
+* `group-ownership` — subtypes `ownership` relation type, relates `owned-group` as `owned` and `group-owner` as `owner`.
+* `object-ownership` — subtypes `ownership` relation, relates `owned-object` as `owned` and `object-owner` as `owner`.
+* `access` — subtypes the `relation` base type, relates `accessed-object` (e.g., files) and `valid-actions` (e.g., 
+  read), plays the role of `requested-change` in `change-request` relation type.
+* `change-request` — subtypes the `relation` base type, relates `requesting-subject`, `requested-subject` and
+  `requested-change`.
+
+### Define rules
+
+#### Overview
+
+Rules are defined independently of any types.
+
+They are executed only as a part of get queries when the [inference](06-infer.md) option is **enabled**. The results of 
+rules execution exist only within the transaction they run in. They are not persisted, and any data inferred in the 
+transaction ceases to exist when the transaction is closed. Read more on rules in the [Rules](#rules) section.
+
+#### Syntax
+
+Rules are defined in TypeQL with the following syntax:
+
+<!-- test-ignore -->
+```typeql
+rule <label>:
+when {
+    <pattern>  [(<pattern>)...]
+} then {
+    <pattern>
+};
+```
+
+The `rule`, `when`, and `then` keywords are specific to rule definitions.
+
+<div class="note">
+[Important]
+Unlike other `define` statements for schema definitions, the rule syntax uses patterns consisting of variables and 
+constraints for **data** instances. For more information see the [Pattern syntax](03-match.md#pattern-syntax) section.
+</div>
+
+#### Rule Validation
+
+The `when` clause (conclusion) of a rule can be a multi statement pattern and can include disjunctions and negations, 
+whereas the `then` clause (condition) should describe a single relation or constraint of ownership of an attribute 
+(due to [Horn-clause logic](https://en.wikipedia.org/wiki/Horn_clause)).
+
+When using a disjunction in a rule, the disjunctive parts must be bound by variables outside of the `or` statement. 
+These variables are the only ones permitted that can be used in the `then` clause.
+
+The `then` clause of a rule can’t use variables that aren’t defined in the `when` clause.
+
+The result of the `then` clause shall not invalidate the pattern, used in the `when` clause of the same rule. Be 
+carefull with negations. Placing the `then` clause statement in the `when` clause is not only redundant (rule will 
+not create a virtual data instance if it already exists, real or virtual one. There is no need to check for it not 
+existing in a rule).
```

Comment:
This isn't precise enough.
You do not need the exact pattern in the then to be in the when for it to be a negative cycle (and violate stratified negation: https://en.wikipedia.org/wiki/Stratification_(mathematics) ). 
You also can have rules contradicting themselves through other rules.

e.g.
``` 
rule friends-if-not-enemies:
when { (...);
  not{  ($x,$y) isa enemies;  }; 
} then { 
 ($x, $y) isa friends;
};

rule enemies-if-not-friends:
when {
  (...);
  not{  ($x,$y) isa friends;}; 
} then {
  ($x, $y) isa enemies;
};
```

Reply:
Please suggest an edit or addition to this paragraph. Keep the audience in mind.

Reply:
The `then` clause must not insert any instance which occurs negated in its `when` clause , or in any the when clause of any rule it may trigger. Attempting to define such a (set of) rules will throw an error.
Rules will not create duplicates of instances which are already in the database or have already been inferred. There is no need to check if it already exists in a rule.